### PR TITLE
CLN: Clean nanops.get_corr_func

### PR DESCRIPTION
**Pull request by dsaxton**
Small cleaning (instead of creating a dictionary of functions and then returning only one, just return the one)

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -1332,31 +1332,24 @@ def nancorr(
...
+    if method == "kendall":
+        from scipy.stats import kendalltau
+
+        return lambda a, b: kendalltau(a, b)[0]
+    elif method == "spearman":
+        from scipy.stats import spearmanr
+
+        return lambda a, b: spearmanr(a, b)[0]
+    elif method == "pearson":
+        return lambda a, b: np.corrcoef(a, b)[0, 1]
```
Comment (jbrockmendel) [inline pandas/core/nanops.py:1344]: can we do this without the lambda?  makes for nicer tracebacks/profiling etc

Comment (dsaxton) [reply pandas/core/nanops.py]: Updated

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -1332,31 +1332,34 @@ def nancorr(
...
-            f"Unknown method '{method}', expected one of 'kendall', 'spearman'"
+            f"Unknown method '{method}', expected one of "
+            "'kendall', 'spearman', 'pearson', or callable"
         )
 
-    def _pearson(a, b):
-        return np.corrcoef(a, b)[0, 1]
-
-    def _kendall(a, b):
-        # kendallttau returns a tuple of the tau statistic and pvalue
```
Comment (jbrockmendel) [inline pandas/core/nanops.py:1350]: is this comment worth retaining?

Comment (dsaxton) [reply pandas/core/nanops.py:1350]: Might as well, should we also have a comment for spearmanr and np.corrcoef?

File: pandas/core/nanops.py [hunk 3]
```diff
@@ -1332,31 +1332,34 @@ def nancorr(
...
-        from scipy.stats import kendalltau, spearmanr
-    elif method in ["pearson"]:
-        pass
+    if method == "kendall":
+        from scipy.stats import kendalltau
+
+        def func(a, b):
+            return kendalltau(a, b)[0]
+
+        return func
```
Comment (jreback) [inline pandas/core/nanops.py:1341]: i would make all of these lambda, e.g.(for all 3 cases)
```
return lambda a, b: kendaltau(a, b)[0]
```

File: pandas/core/nanops.py [hunk 4]
```diff
@@ -1332,31 +1332,34 @@ def nancorr(
...
+    elif method == "pearson":
+
+        def func(a, b):
+            return np.corrcoef(a, b)[0, 1]
+
+        return func
     elif callable(method):
         return method
     else:
```
Comment (jreback) [inline pandas/core/nanops.py:1357]: no else needed, just the raise

**Review by jreback — CHANGES_REQUESTED** [review]
do we have tests for invaid method names?

File: pandas/core/nanops.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/nanops.py:1341]: hah, i asked for these to be changed from lambda to non-lambdas for nicer tracebacks

Comment (jreback) [reply pandas/core/nanops.py:1341]: oh really it does? hmm, ok, then that's fine

**Comment by dsaxton** [comment]
> do we have tests for invaid method names?

There's testing at least for DataFrame.corr (test_corr_invalid_method in /tests/frame/methods/test_cov_corr.py) but not this function directly AFAIK

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**